src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2955,7 +2965,12 @@ protected function asDateTime($value)
         // Finally, we will just assume this date is in the format used by default on
         // the database connection and use that format to create the Carbon object
         // that is returned back out to the developers after we convert it here.
-        return Carbon::createFromFormat($this->getDateFormat(), $value);
+        if (static::$timezone) {
+            return Carbon::createFromFormat($this->getDateFormat(), $value, static::$timezone)
+                ->setTimezone(date_default_timezone_get());
```

Comment:
The date as stored and retrieved from the database is NOT affected by the PHP default timezone. We are just converting the date, after being retrieved, into the PHP default timezone, which is probably what the programmer expects.